openshift/oc-mirror
# OCPBUGS-7465: Fix 401 Unauthorized due to stale auth token

# Description

Creating this PR with changes made by @grokspawn (originally started this work but had to transition to focus on other things so I took over) that:
- refactors the conditional code compilation to use go build tags to ignore building for windows (necessary for being able to compile on MacOS)
- bumps the operator-framework/operator-registry dependency to v1.29.0 and fixes necessary unit tests

Fixes #526 

## Type of change

Please delete options that are not relevant.


# How Has This Been Tested?

This was tested by manually trying to reproduce the problem found in #526 by performing the following steps:
1. `./hack/build.sh` to build a binary with the changes
2. Create a local image registry with `docker run -d -p 5001:5000 --restart=always --name registry registry`
3. Create an oc-mirror config file named `oc-mirror-config.yaml` with contents:
```yaml
apiVersion: mirror.openshift.io/v1alpha2
kind: ImageSetConfiguration
storageConfig:
  registry:
    imageURL: localhost:5001/metadata:latest
    skipTLS: true
mirror:
  operators:
    - catalog: registry.redhat.io/redhat/redhat-operator-index:v4.11
      packages:
        - name: loki-operator
        - name: elasticsearch-operator
        - name: odf-operator
```
4. Running the oc-mirror binary with the changes:
```
./bin/oc-mirror --config ~/testing/oc-mirror/oc-mirror-config.yaml docker://localhost:5001 --dest-skip-tls --dest-use-http --verbose 9 2>&1 | tee /tmp/oc-mirror-attempt.txt
```

Checking the output from step 4 a second 401 unauthorized response is encountered and then 200 OK statuses are retrieved in subsequent requests and the command exits successfully.

> **Note**: This was a preliminary test and more testing is being done to verify this effectively resolves the problem. This PR is being opened as a draft to allow for others to more easily run tests for verification.

**Test Configuration**:
* Firmware version:
* Hardware:
* Toolchain:
* SDK:

# Checklist:
